### ISSUE TITLE
vinyl: remove VINYL_KEY_MAXLEN limit

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#1604]
Vinyl doesn't allow keys large than 1024 bytes. I tried to remove this check from `vinyl_tuple_new()` and realized that vinyl uses uint16_t numbers to calculate sizes:

```
struct PACKED sdindexpage {
	uint64_t offset;
	uint32_t offsetindex;
	uint32_t size;
	uint32_t sizeorigin;
	uint16_t sizemin; <-- OMG
	uint16_t sizemax;  <-- OMG
	uint64_t lsnmin;
	uint64_t lsnmax;
};
```

**Comment by rtsisyk** [comment]
Probably fixed by 0aa81d87950069ea3f4eeba077e3fe8ec45b6507, please add remove limit and add a test case.

**Comment by rtsisyk** [comment]
See also #1530 - assertion is  same.
